Resources/public/js/views/services/plugins/ez-locationswapplugin.js

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-locationswapplugin', function (Y) {
+    "use strict";
+    /**
+     * Provides the plugin for swap location
+     *
+     * @module ez-locationswapplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     * Swap location plugin. It sets an event handler to swap location
+     *
+     * In order to use it you need to fire `swapLocation` event with parameter
+     * `location` containing the eZ.Location object you want swap with other.
+     *
+     * @namespace eZ.Plugin
+     * @class LocationSwap
+     * @constructor
+     * @extends eZ.Plugin.ViewServiceBase
+     */
+    Y.eZ.Plugin.LocationSwap = Y.Base.create('locationswapplugin', Y.eZ.Plugin.ViewServiceBase, [], {
+
+        initializer: function () {
+            this.onHostEvent('*:swapLocation', this._swapLocationSelect);
+        },
+
+        /**
+         * swapLocation event handler, launch the universal discovery widget
+         * to choose the location you want to swap with the given one
+         *
+         * @method _swapLocationSelect
+         * @private
+         * @param {EventFacade} e swapLocation event facade
+         */
+        _swapLocationSelect: function (e) {
+            var service = this.get('host');
+
+            service.fire('contentDiscover', {
+                config: {
+                    title: "Select the location you want to swap with the given one",
+                    contentDiscoveredHandler: Y.bind(this._swapLocation, this),
+                    multiple: false,
+                    isSelectable: true,
+                    data: {
+                        afterSwapLocationCallback: e.afterSwapLocationCallback
+                    }
+                },
+            });
+        },
+
+        /**
+         * Swaps the given location with the selected one
+         *
+         * @method _swapLocation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _swapLocation: function (e) {
+            var service = this.get('host'),
+                capi = service.get('capi'),
+                location = service.get('location'),
+                notificationIdentifier = 'swap-location-' + location.get('id'),
+                data = e.target.get('data'),
+                destinationLocation = e.selection.location,
+                errNotificationIdentifier = 'swap-location-' + location.get('id'),
+                locationSwaped = false,
+                stack = new Y.Parallel(),
```

Comment:
since you only have one asynchronous operation, the use of Y.Parallel is not needed. You should write something like:

```js
location.swap({api: capi}, destinationLocation, function (error) {
    if ( error ) {
        // error notification
    } else {
        // success notification
    }
});
```